pallets/connectors/src/lib.rs

```diff
@@ -0,0 +1,507 @@
+			Self::deposit_event(Event::SetDomainRouter { domain, router });
+
+			Ok(())
+		}
+
+		/// Add a pool to a given domain
+		#[pallet::weight(< T as Config >::WeightInfo::add_pool())]
+		pub fn add_pool(
+			origin: OriginFor<T>,
+			pool_id: PoolIdOf<T>,
+			domain: Domain,
+		) -> DispatchResult {
+			let who = ensure_signed(origin.clone())?;
+
+			ensure!(
+				T::PoolInspect::pool_exists(pool_id),
+				Error::<T>::PoolNotFound
+			);
+
+			Self::do_send_message(who, Message::AddPool { pool_id }, domain)?;
+
+			Ok(())
+		}
+
+		/// Add a tranche to a given domain
+		#[pallet::weight(< T as Config >::WeightInfo::add_tranche())]
+		pub fn add_tranche(
+			origin: OriginFor<T>,
+			pool_id: PoolIdOf<T>,
+			tranche_id: TrancheIdOf<T>,
+			domain: Domain,
+		) -> DispatchResult {
+			let who = ensure_signed(origin.clone())?;
+
+			ensure!(
+				T::PoolInspect::tranche_exists(pool_id, tranche_id),
+				Error::<T>::TrancheNotFound
+			);
+
+			// Send the message through the router
+			//
+			// TODO: retrieve token name and symbol from asset-registry
+			// Depends on https://github.com/centrifuge/centrifuge-chain/issues/842
+			//
+			// TODO: only allow calling add_tranche when
+			// both the name and symbol are non-zero values.
+			Self::do_send_message(
+				who,
+				Message::AddTranche {
+					pool_id,
+					tranche_id,
+					token_name: [0; 32],
+					token_symbol: [0; 32],
+				},
+				domain,
+			)?;
+
+			Ok(())
+		}
+
+		/// Update a token price
+		#[pallet::weight(< T as Config >::WeightInfo::update_token_price())]
+		pub fn update_token_price(
+			origin: OriginFor<T>,
+			pool_id: PoolIdOf<T>,
+			tranche_id: TrancheIdOf<T>,
+			domain: Domain,
+		) -> DispatchResult {
+			let who = ensure_signed(origin.clone())?;
+
+			ensure!(
+				T::PoolInspect::tranche_exists(pool_id, tranche_id),
+				Error::<T>::TrancheNotFound
+			);
+
+			let latest_price = T::PoolInspect::get_tranche_token_price(pool_id, tranche_id)
+				.ok_or(Error::<T>::MissingPrice)?;
+
+			Self::do_send_message(
+				who,
+				Message::UpdateTokenPrice {
+					pool_id,
+					tranche_id,
+					price: latest_price.price,
+				},
+				domain,
+			)?;
+
+			Ok(())
+		}
+
+		/// Update a member
+		#[pallet::weight(< T as Config >::WeightInfo::update_member())]
+		pub fn update_member(
+			origin: OriginFor<T>,
+			address: DomainAddress<Domain>,
+			pool_id: PoolIdOf<T>,
+			tranche_id: TrancheIdOf<T>,
+			valid_until: Moment,
+		) -> DispatchResult {
+			let who = ensure_signed(origin.clone())?;
+
+			// Check that the origin is a member of this tranche token or
+			// is a MemberListAdmin and thus allowed to add other members.
+			ensure!(
+				T::Permission::has(
+					PermissionScope::Pool(pool_id),
+					who.clone(),
+					Role::PoolRole(PoolRole::TrancheInvestor(tranche_id, Self::now()))
+				) || T::Permission::has(
+					PermissionScope::Pool(pool_id),
+					who.clone(),
+					Role::PoolRole(PoolRole::MemberListAdmin)
+				),
+				BadOrigin
+			);
```

Comment:
Does this mean anybody who is a tranche investor is allowed to add a domain as a tranche investor? This seems wrong to me.

Reply:
This was coded by @offerijns but my understanding is not that we are adding a Domain as a tranche investor but rather that we are sending an `UpdateMember` message to update the relative state on the target Domain. Does that make sense?

Reply:
The idea was to use the permission pallet to bookkeep whitelisted addresses on other domains (chains), by storing a derived account from a `DomainAddress`, which represents an address on another domain.

The reasoning for having this `or` clause was that if you are already a whitelisted tranche investor on centrifuge chain, you might be able to whitelist your own address on another domain. But I've been going back and forth on whether this makes sense as well, so for now I think it would probably be cleaner to just only allow `MemberListAdmin`s to whitelist `DomainAddress`ses.

Reply:
Is the whitelisted address from derived from the user account that we have here on-chain? I.e. can we link Centrifuge account x to Ethereum account y? Else, we would give users who are tranche investors `MemberListAdmin` rights on all other chains we connect to, right?

Reply:
Yeah this is exactly the problem. I think for now we should simplify this to only allowing `MemberListAdmin`s to add members, so removing the OR clause here (cc @NunoAlexandre). But maybe there's value in having links between Centrifuge accounts and other domain accounts, so let's have another discussion on this when I'm back.

Reply:
Sound good @offerijns, thanks!